src/main/webapp/WEB-INF/tags/student/comments/commentsForStudentsPanel.tag

```diff
@@ -0,0 +1,34 @@
+<%@ tag description="Comments for students" %>
+<%@ taglib uri="http://java.sun.com/jsp/jstl/core" prefix="c" %>
+<%@ taglib tagdir="/WEB-INF/tags/student/comments" prefix="comments" %>
+<%@ attribute name="commentRows" type="java.util.Collection" required="true" %>
+<div class="panel panel-primary">
+    <div class="panel-heading">
+        <strong>Comments for students</strong>
+    </div>
+    <div class="panel-body">
+        <c:set var="commentIdx" value="0" />
+        <c:forEach items="${commentRows}" var="commentRow" varStatus="i"> <%-- comment loop starts --%>
+            <c:set var="studentIdx" value="${i.index + 1}" />
+            <div class="panel panel-info student-record-comments ${commentRow.recipientDetails == 'you' ? 'giver_display-to-you' : 'giver_display-to-others'}">
+                <div class="panel-heading">
+                    To <b>${commentRow.recipientDetails}</b>
+                </div>
+                <ul class="list-group comments">
+                    <c:set var="commentIdx" value="${commentIdx + 1}" />
```

Comment:
My bad here, `commentIdx` here should be `i.index + 1`, and `studentIdx` isn't used at all. I've removed `studentIdx` and set `commentIdx` appropriately